transport/tchannel/handler_test.go

```diff
@@ -273,6 +274,17 @@ func TestResponseWriter(t *testing.T) {
 			arg2: []byte("{}\n"),
 			arg3: []byte("{}"),
 		},
+		{
+			format: tchannel.Raw,
+			apply: func(w *responseWriter) {
+				w.SetApplicationError()
```

Comment:
Yes. This is API exposed by the transports to the encodings. Transport implementors may implement this API to behave differently in case of application errors. TChannel sets the failure bit if this is called.